### PR TITLE
on_action user var name (#368)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
retrieve original var name for on_action

### CODE REVIEW

File: src/taipy/gui/utils/types.py [hunk 1]
```diff
@@ -103,28 +103,31 @@ def cast_value(self, value: t.Any):
...
+        return _HOLDER_PREFIX_START + "C"
 
 
 class _TaipyContentImage(_TaipyBase):
     @staticmethod
     def get_hash():
-        return "_TpCi"
+        return _HOLDER_PREFIX_START + "Ci"
+
+_HOLDER_PREFIX_START = "_Tp"
```
Comment (FabienLelaquais) [inline src/taipy/gui/utils/types.py]: Is there a matching HOLDER_PREFIX_END or something?
PREFIX really means... starting a string value of some sort.
I mean HOLDER_PREFIX should suffice.

File: src/taipy/gui/utils/types.py [hunk 2]
```diff
@@ -45,13 +45,13 @@ def _get_readable_name(self):
 
     @staticmethod
     def get_hash():
-        return "TaipyBase"
+        return NotImplemented
```
Comment (FabienLelaquais) [inline src/taipy/gui/utils/types.py:48]: Shouldn't this be a raise instead of a return?

File: src/taipy/gui/gui.py [hunk 3]
```diff
@@ -445,18 +452,24 @@ def __call_on_change(self, var_name: str, value: t.Any, on_change: t.Optional[st
             var_name = var_name_decode
         else:
             if var_name not in self.__var_dir._var_head:
-                warnings.warn(f"Can't find matching variable for {var_name} on {current_context} context")
-                return
+                raise NameError(f"Can't find matching variable for {var_name} on {current_context} context")
```
Comment (FabienLelaquais) [inline src/taipy/gui/gui.py]: I'm not sure but it seems like 'on context: {current_context}' might be easier to read

File: src/taipy/gui/gui.py [hunk 4]
```diff
@@ -445,18 +452,24 @@ def __call_on_change(self, var_name: str, value: t.Any, on_change: t.Optional[st
...
             for k, v in self.__var_dir._var_head[var_name]:
                 if v == current_context:
                     var_name = k
                     _found = True
                     break
             if not _found:
-                warnings.warn(f"Can't find matching variable for {var_name} on {current_context} context")
-                return
-        var_name = f"{var_name}.{suffix_var_name}" if suffix_var_name else var_name
+                raise NameError(f"Can't find matching variable for {var_name} on {current_context} context")
```
Comment (FabienLelaquais) [inline src/taipy/gui/gui.py]: Same remark on error message

File: src/taipy/gui/gui.py [hunk 5]
```diff
@@ -445,18 +452,24 @@ def __call_on_change(self, var_name: str, value: t.Any, on_change: t.Optional[st
...
+        return f"{var_name}.{suffix_var_name}" if suffix_var_name else var_name, current_context
+
+
+    def __call_on_change(self, var_name: str, value: t.Any, on_change: t.Optional[str] = None):
+        try:
+            var_name, current_context = self.__get_real_var_name(var_name)
+        except Exception as e:
+            warnings.warn(f"{e}")
```
Comment (FabienLelaquais) [inline src/taipy/gui/gui.py:471]: Is a warning enough at this point? May be an actual error?

**Review by FabienLelaquais — APPROVED** [review]
Minor changes, The actual issue seems to be properly handled. Well done!

File: src/taipy/gui/utils/types.py (hunk 1 above)
Comment (FredLL-Avaiga) [reply src/taipy/gui/utils/types.py]: true

File: src/taipy/gui/gui.py (hunk 3 above)
Comment (FredLL-Avaiga) [reply src/taipy/gui/gui.py]: ok

File: src/taipy/gui/gui.py (hunk 5 above)
Comment (FredLL-Avaiga) [reply src/taipy/gui/gui.py:471]: I didn't touch this part (only refactor)

File: src/taipy/gui/utils/types.py (hunk 2 above)
Comment (FredLL-Avaiga) [reply src/taipy/gui/utils/types.py:48]: no I think you can return NotImplemented or raise NotImplementedError
https://stackoverflow.com/questions/878943/why-return-notimplemented-instead-of-raising-notimplementederror

**Review by dinhlongviolin1 — APPROVED** [review]
lgtm

**Pull request merged**